### PR TITLE
add all pages of order.

### PR DESCRIPTION
**Pull request by Alice1319**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -60,5 +60,8 @@
         "eslint-plugin-require-explicit-generics": "^0.4.2",
         "prettier": "2.8.4",
         "typescript": "^4.9.5"
-    }
+    },
+    "eslintIgnore": [
```
Comment (swaroopar) [inline package.json:64]: this must be removed.

File: src/components/content/order/CreateService.tsx [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+    Ocl,
+    RegisterServiceEntity,
+} from '../../../xpanse-api/generated';
+import { OrderSubmitProps } from './OrderSubmit';
+import { DeployParam } from './orderInputItem/OrderCommon';
+import Navigate from './Navigate';
+import { Tab } from 'rc-tabs/lib/interface';
+import { HuaWeiLogo, AzureLogo, AlibabaLogo, AWSLogo, XpanseLogo } from './CSPLogo';
+
+interface CSP {
```
Comment (swaroopar) [inline src/components/content/order/CreateService.tsx:25]: please move it to a separate class.

File: src/components/content/order/CreateService.tsx [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+    name: string;
+    icon?: string;
+    logo?: string;
+}
+
+const cspMap: Map<CloudServiceProviderNameEnum, CSP> = new Map([
```
Comment (swaroopar) [inline src/components/content/order/CreateService.tsx:31]: also to a separate file.

File: src/components/content/order/CreateService.tsx [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+                setVersionValue(versions[0].value);
+                updateArea(versions[0].value, result);
+            } else {
+                return;
+            }
+        });
+    }, [location]);
+
+    //set csp
+    function updateArea(version: string, mapper: Map<string, RegisterServiceEntity[]>): void {
```
Comment (swaroopar) [inline src/components/content/order/CreateService.tsx:129]: function name does not reflect the tasks performed by function. 
Also if possible, please split this function.

File: src/components/content/order/CSPLogo.tsx [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import HuaWeiLogo from '../../../img/Huawei_Logo.png';
```
Comment (swaroopar) [inline src/components/content/order/CSPLogo.tsx:1]: rename the file to CspLogo.tsx

File: src/components/content/home/WelcomeCard.tsx [hunk 6]
```diff
@@ -5,17 +5,20 @@
 
 import { RocketOutlined, SettingOutlined } from '@ant-design/icons';
 import { Card, Divider } from 'antd';
+import Navigate from '../order/Navigate';
```
Comment (swaroopar) [inline src/components/content/home/WelcomeCard.tsx:8]: please fix all ES lint warnings

**Review by swaroopar — CHANGES_REQUESTED** [review]
Have written some initial comments. but cannot check further because the API changes which re used here are not yet merged. We need those changes to be merged first.

File: src/components/content/home/WelcomeCard.tsx (hunk 6 above)
Comment (Alice1319) [reply src/components/content/home/WelcomeCard.tsx:8]: fixed.

File: src/components/content/register/DisplayOclData.tsx [hunk 7]
```diff
@@ -77,23 +77,26 @@ function DisplayOclData({ ocl }: { ocl: Ocl }): JSX.Element | string {
...
                                 <br />
                                 <br />
                             </div>
                             <div>
                                 <b>Available Regions</b>
                                 <br />
-                                {ocl.cloudServiceProvider && ocl.cloudServiceProvider.regions
-                                    ? ocl.cloudServiceProvider.regions.join(', ')
-                                    : PLACE_HOLDER_UNKNOWN_VALUE}
+                                {ocl.cloudServiceProvider.areas.map((area, index) => (
```
[the remark this answers is not in the document]
Comment (iskey) [reply src/components/content/register/DisplayOclData.tsx:94]: I suggest making the area a tag/property of the region, not the father of the region.

**Comment by iskey** [comment]
> Have written some initial comments. but cannot check further because the API changes which re used here are not yet merged. We need those changes to be merged first.

yes, agree, we must let the area part be merged in first. This PR should be split into different PRs.

**Review by iskey — CHANGES_REQUESTED** [review]
please, refactor the service select page.

File: package.json (hunk 1 above)
Comment (Alice1319) [reply package.json]: fixed.

File: src/components/content/order/CreateService.tsx (hunk 2 above)
Comment (Alice1319) [reply src/components/content/order/CreateService.tsx]: fixed.

File: src/components/content/order/CreateService.tsx (hunk 4 above)
Comment (Alice1319) [reply src/components/content/order/CreateService.tsx]: fixed.

File: src/components/content/register/DisplayOclData.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (Alice1319) [reply src/components/content/register/DisplayOclData.tsx]: fixed.

File: src/components/content/order/CreateService.tsx (hunk 3 above)
Comment (Alice1319) [reply src/components/content/order/CreateService.tsx]: fixed.

**Comment by iskey** [comment]
What I meant is to reset this branch to our co-development branch, the commit history is not clear here (including some commits like `Alice1319@users.noreply.github.com`). And also, it seems your ECA did not sign correctly. I will create another PR for this, please move #10.

**Pull request closed without merging**